nfs-ha-cluster-ubuntu/azuredeploy.json

```diff
@@ -0,0 +1,229 @@
+            },
+            "defaultValue": 1,
+            "minValue": 1,
+            "maxValue": 8,
+            "type": "int"
+        },
+        "dataDiskSizeInGB": {
+            "defaultValue": 32,
+            "metadata": {
+                "description": "Size per disk in an NFS server"
+            },
+            "type": "int"
+        },
+        "resourcesUniqueString": {
+            "metadata": {
+                "description": "Unique string of fixed length (e.g., 6) identifying related resources"
+            },
+            "type": "string",
+            "defaultValue": "[substring(uniqueString(resourceGroup().id, deployment().name), 3, 6)]"
+        }
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2017-05-10",
+            "name": "pid-38a42e4d-89db-4159-9b75-9113b6eae9b6",
+            "properties": {
+                "mode": "Incremental",
+                "template": {
+                    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+                    "contentVersion": "1.0.0.0",
+                    "resources": []
+                }
+            }
+        },
+        {
+            "condition": "[equals(parameters('subnetId'), 'create-for-ci')]",
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2017-05-10",
+            "name": "nfs-ha-vnet-default-deployment",
+            "properties": {
+                "mode": "Incremental",
+                "parameters": {
+                    "location": {
+                        "value": "[parameters('location')]"
+                    }
+                },
+                "templateLink": {
+                    "uri": "[concat(parameters('_artifactsLocation'), 'nested/nfs-ha-vnet-default.json', parameters('_artifactsLocationSasToken'))]"
+                }
+            }
+        },
+        {
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2017-05-10",
+            "dependsOn": [
+                "Microsoft.Resources/deployments/nfs-ha-vnet-default-deployment"
+            ],
+            "name": "nfs-ha-deployment",
+            "properties": {
+                "mode": "Incremental",
+                "parameters": {
+                    "_artifactsLocation": {
+                        "value": "[parameters('_artifactsLocation')]"
+                    },
+                    "_artifactsLocationSasToken": {
+                        "value": "[parameters('_artifactsLocationSasToken')]"
+                    },
+                    "location": {
+                        "value": "[parameters('location')]"
+                    },
+                    "subnetId": {
+                        "value": "[if(equals(parameters('subnetId'), 'create-for-ci'), reference('nfs-ha-vnet-default-deployment').outputs.subnetId.value, parameters('subnetId'))]"
```

Comment:
let's remove this pattern and rely on the prereqs for creating and existing resources that are needed - that will simplify for customer that want to adopt this

Reply:
Done as commented. Thanks.